osu.Game/Rulesets/Scoring/ScoreProcessor.cs

```diff
@@ -98,7 +104,11 @@ public abstract class ScoreProcessor
         protected ScoreProcessor()
         {
             Combo.ValueChanged += delegate { HighestCombo.Value = Math.Max(HighestCombo.Value, Combo.Value); };
-            Accuracy.ValueChanged += delegate { Rank.Value = rankFrom(Accuracy.Value); };
+            Accuracy.ValueChanged += delegate
+            {
+                foreach (var mod in Mods.Value.OfType<IApplicableToScoreProcessor>())
+                    Rank.Value = mod.AdjustRank(rankFrom(Accuracy.Value), Accuracy.Value);
```

Comment:
shouldn't the Rank.Value be passed along each iteration as to allow chaining from multiple mods?